sdgym/dataset_explorer.py

```diff
@@ -0,0 +1,270 @@
+        Returns:
+            int:
+                The maximum number of children linked to a single parent table.
+        """
+        branch_counts = defaultdict(set)
+        for rel in relationships:
+            parent = rel['parent_table_name']
+            child = rel['child_table_name']
+            branch_counts[parent].add(child)
+
+        return max((len(children) for children in branch_counts.values()), default=0)
+
+    @staticmethod
+    def _get_max_depth(metadata):
+        """Calculate the maximum depth of a metadata.
+
+        Args:
+            metadata (sdv.metadata.Metadata):
+                The SDV Metadata object representing the dataset.
+
+        Returns:
+            int:
+                The maximum schema depth (i.e., the longest parent-child relationship chain).
+        """
+        child_map = metadata._get_child_map()
+        parent_map = metadata._get_parent_map()
+        if not child_map:
+            return 1
+
+        def dfs(table):
+            if table not in child_map or not child_map[table]:
+                return 1
+
+            return 1 + max(dfs(child) for child in child_map[table])
+
+        parent_map = metadata._get_parent_map()
+        root_tables = [table for table in child_map.keys() if table not in parent_map]
+        if not root_tables:
+            return 1
+
+        return max(dfs(root) for root in root_tables)
+
+    @staticmethod
+    def _summarize_metadata_columns(metadata):
+        """Summarize column-level details from a dataset’s metadata.
+
+        Args:
+            metadata (sdv.metadata.Metadata):
+                The SDV Metadata object containing table and column information.
+
+        Returns:
+            dict:
+                A dictionary summarizing total and per-type column counts across all tables.
+        """
+        results = {
+            'Total_Num_Columns': 0,
+            'Total_Num_Columns_Categorical': 0,
+            'Total_Num_Columns_Numerical': 0,
+            'Total_Num_Columns_Datetime': 0,
+            'Total_Num_Columns_PII': 0,
+            'Total_Num_Columns_ID_NonKey': 0,
+            'Max_Num_Columns_Per_Table': 0,
+        }
+        for table_name, table in metadata.tables.items():
+            num_cols = len(table.columns)
+            results['Total_Num_Columns'] += num_cols
+            results['Max_Num_Columns_Per_Table'] = max(
+                results['Max_Num_Columns_Per_Table'], num_cols
+            )
+            for column_name, column in table.columns.items():
+                sdtype = column['sdtype']
+                if sdtype in ['categorical', 'boolean']:
+                    results['Total_Num_Columns_Categorical'] += 1
+                elif sdtype in ['numerical']:
+                    results['Total_Num_Columns_Numerical'] += 1
+                elif sdtype in ['datetime']:
+                    results['Total_Num_Columns_Datetime'] += 1
+                elif sdtype in ['id'] and column_name != table.primary_key:
```

Comment:
The documentation mentions it shouldn't be any type of key, not just primary key.

Reply:
Done here [40b0b83](https://github.com/sdv-dev/SDGym/pull/472/commits/40b0b83e54042bf86591f507711826b393c392ea)